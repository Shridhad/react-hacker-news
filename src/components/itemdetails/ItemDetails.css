

.item-details .list-item {
    padding-left: 0;
    border-bottom: none; 
}

.item-details .item-content,
.item-details .item-comments {
    padding-left: 1rem;
}