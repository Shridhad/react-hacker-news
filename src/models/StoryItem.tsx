import StoryComment from './StoryComment';

export default interface StoryItem {
    id: number,
    title: string,
    url: string,
    user: string,
    time_ago: string,
    points: number,
    content?: string,
    comments_count: number,
    comments?: Array<StoryComment>
};
