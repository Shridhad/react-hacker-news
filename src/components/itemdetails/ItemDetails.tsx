import React from 'react';
import Comment from '../comment/Comment';
import ListItem from '../listItem/ListItem';
import StoryItem from '../../models/StoryItem';
import StoryComment from '../../models/StoryComment';

import './ItemDetails.css';

export default class ItemDetails extends React.Component<StoryItem, {}> {
    render() {
        return (
            <div className="item-details">
                <ListItem {...this.props}></ListItem>
                <div className="item-content">{this.props.content}</div>

                <div className="item-comments">
                    {this.renderComments(this.props.comments)}
                </div>
            </div>
        )
    }

    renderComments(comments: Array<StoryComment> = []) {
        return comments.map(comment => <Comment {...comment} key={comment.id}/>);
    }
}