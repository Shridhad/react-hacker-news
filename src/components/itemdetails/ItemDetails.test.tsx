import React from 'react';
import {shallow} from 'enzyme';
import ItemDetails from './ItemDetails';
import StoryItem from '../../models/StoryItem';

describe("<ItemDetails />", () => {
    it("should render list item", () => {
        const story = fetchStory();

        const details = shallow(<ItemDetails {...story} />);
        const listItems = details.find('ListItem');

        expect(listItems.length).toBe(1);
    });

    it("should render story item content", () => {
        const story = fetchStory(true);
        const details = shallow(<ItemDetails {...story} />);
        
        const content = details.find('.item-content');
        expect(content.length).toBe(1);
        expect(content.text()).toEqual(story.content);
    });

    it("should render comments", () => {
        const story = fetchStory(false, true);

        const details = shallow(<ItemDetails {...story} />);
        const comments = details.find('Comment');

        expect(comments.length).toBe(story.comments_count);
    });
});

const fetchStory = (withContent = false, withComments = false) => {
    const story: StoryItem = {
        "id": 19780387,
        "title": "Topics in Advanced Data Structures [pdf]",
        "points": 131,
        "user": "htiek",
        "time_ago": "2 hours ago",
        "comments_count": 2,
        "url": "http://web.stanford.edu/class/cs166/handouts/100%20Suggested%20Final%20Project%20Topics.pdf",
    };

    if (withContent) {
        story.content = "I set out to build a reverse address book.";
    }

    if (withComments) {
        story.comments = [
            {
                "id": 19790143,
                "user": "0xab",
                "time_ago": "18 minutes ago",
                "content": "<p>I do research in computer vision and this paper is so bad it",
                "comments_count": 1,
                "level": 1,
                "url": "item?id=19790249"
            },
            {
                "id": 19790143,
                "user": "0xab",
                "time_ago": "18 minutes ago",
                "content": "<p>I do research in computer vision and this paper is so bad it",
                "comments_count": 1,
                "level": 1,
                "url": "item?id=19790249"
            }
        ]
    }

    return story;
}
