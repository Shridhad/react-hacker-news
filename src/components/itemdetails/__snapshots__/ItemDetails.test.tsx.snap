// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemDetails /> should render list item 1`] = `ShallowWrapper {}`;
